ui/app/src/views/ViewDashboard.tsx

```diff
@@ -11,32 +11,78 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { useQueryParams, StringParam } from 'use-query-params';
-import { DashboardResource } from '@perses-dev/core';
 import { ViewDashboard as DashboardView } from '@perses-dev/dashboards';
+import { useParams } from 'react-router-dom';
+import { Box, useTheme } from '@mui/material';
+import {
+  ChartsThemeProvider,
+  ErrorAlert,
+  ErrorBoundary,
+  generateChartsTheme,
+  PersesChartsTheme,
+} from '@perses-dev/components';
+import { useMemo } from 'react';
+import { PluginRegistry } from '@perses-dev/plugin-system';
+import { useBundledPlugins } from '../model/bundled-plugins';
+import { useDashboard } from '../model/dashboard-client';
 import { useDatasourceApi } from '../model/datasource-api';
-import { useSampleData } from '../utils/temp-sample-data';
 
-const DEFAULT_DASHBOARD_ID = 'node-exporter-full';
+// app specific echarts option overrides, empty since perses uses default
+// https://apache.github.io/echarts-handbook/en/concepts/style/#theme
+const ECHARTS_THEME_OVERRIDES = {};
+
+interface ViewSingleDashboardProperty {
+  dashboardName: string;
+  projectName: string;
+}
+
+function ViewSingleDashboard(props: ViewSingleDashboardProperty) {
+  const { dashboardName, projectName } = props;
+  // TODO: Loading indicator
+  const { data, isLoading } = useDashboard(projectName, dashboardName);
+  const datasourceApi = useDatasourceApi();
+
+  if (!isLoading && data !== undefined) {
+    return <DashboardView dashboardResource={data} datasourceApi={datasourceApi} />;
+  }
+  return null;
+}
 
 /**
  * The View for viewing a Dashboard.
  */
 function ViewDashboard() {
-  const [query] = useQueryParams({ dashboard: StringParam });
-  const { dashboard } = query;
-
-  // TODO: setup project routing
-  const sampleDashboard = useSampleData<DashboardResource>(dashboard || DEFAULT_DASHBOARD_ID);
+  const { getInstalledPlugins, importPluginModule } = useBundledPlugins();
+  const muiTheme = useTheme();
+  const chartsTheme: PersesChartsTheme = useMemo(() => {
+    return generateChartsTheme('perses', muiTheme, ECHARTS_THEME_OVERRIDES);
+  }, [muiTheme]);
 
-  const datasourceApi = useDatasourceApi();
-
-  // TODO: Loading indicator
-  if (sampleDashboard === undefined) {
+  const projectName = useParams().projectID;
+  const dashboardName = useParams().dashboardID;
```

Comment:
Feels weird that we're switching back and forth between calling these "xxxName" and "xxxID". Shouldn't they just be one or the other?